File: lib/java/src/main/java/io/github/brenbar/uapi/internal/validation/ValidateBoolean.java
package io.github.brenbar.uapi.internal.validation;

import java.util.List;

import static io.github.brenbar.uapi.internal.types.UBoolean._BOOLEAN_NAME;
import static io.github.brenbar.uapi.internal.validation.GetTypeUnexpectedValidationFailure.getTypeUnexpectedValidationFailure;

public class ValidateBoolean {
    public static List<ValidationFailure> validateBoolean(Object value) {
        if (value instanceof Boolean) {
            return List.of();
        } else {
            return getTypeUnexpectedValidationFailure(List.of(), value, _BOOLEAN_NAME);
        }
    }
}

File: lib/py/uapi/internal/validation/ValidateBoolean.py
from typing import List, TYPE_CHECKING
from uapi.internal.types.UBoolean import _BOOLEAN_NAME

if TYPE_CHECKING:
    from uapi.internal.validation.ValidationFailure import ValidationFailure


def validate_boolean(value: object) -> List['ValidationFailure']:
    from uapi.internal.validation.GetTypeUnexpectedValidationFailure import get_type_unexpected_validation_failure

    if isinstance(value, bool):
        return []
    else:
        return get_type_unexpected_validation_failure([], value, _BOOLEAN_NAME)
